org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/UnusedProblemFactory.java

```diff
@@ -173,6 +174,11 @@ public List<CategorizedProblem> addUnusedPrivateMembers(CompilationUnitTree unit
 				VarSymbol varSymbol = variableDecl.sym;
 				ElementKind varKind = varSymbol == null ? null : varSymbol.getKind();
 				if (varKind == ElementKind.FIELD) {
+					if( varSymbol.owner instanceof ClassSymbol css) {
```

Comment:
This seems like the UnusedTreeScanner returns an incorrect symbol here. Could that check be added directly in UnusedTreeScanner?
One reason is that, somewhere down the road, when we're happy enough with the UnusedTreeScanner enough, we might try to contribute it into OpenJDK so it becomes available with `-Xlint:unused` option. To achieve this, we need the UnusedTreeScanner to be as exhaustive as possible without depending on Eclipse bits.